File: src/main/java/skywolf46/extrautility/ExtraUtilityPlugin.java
package skywolf46.extrautility;

import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;
import skywolf46.extrautility.listeners.DamageListener;
import skywolf46.extrautility.listeners.InteractionListener;

public class ExtraUtilityPlugin extends JavaPlugin {
    private static ExtraUtilityPlugin inst;

    public static ExtraUtilityPlugin inst() {
        return inst;
    }

    @Override
    public void onEnable() {
        inst = this;
        Bukkit.getPluginManager().registerEvents(new DamageListener(), this);
        Bukkit.getPluginManager().registerEvents(new InteractionListener(), this);
    }
}

File: src/main/kotlin/skywolf46/extrautility/ExtraUtilityPlugin.kt
package skywolf46.extrautility

import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin
import skywolf46.extrautility.listener.DamageListener
import skywolf46.extrautility.listener.InteractionListener
import skywolf46.extrautility.util.log

internal lateinit var inst: ExtraUtilityPlugin
    private set
class ExtraUtilityPlugin : JavaPlugin() {

    override fun onEnable() {
        inst = this
        log("§e[ExtraUtility] §7Initializing " + getVersion())
        Bukkit.getPluginManager().registerEvents(DamageListener(), this)
        Bukkit.getPluginManager().registerEvents(InteractionListener(), this)

    }

    fun getVersion(): String? = description.version
}